### ISSUE TITLE
Add a script to automate creating fill invalidations 

### ISSUE THREAD

**Issue by istankovic** [beamer-bridge/beamer#1646]
Currently we have to manually send fill invalidations between supported chains, which is cumbersome and error-prone.
It would be good to have this automated so that we can be more efficient after making a new deployment.
To that end, write a script that

- loads the chain RPC URLs from a file, for all supported chains
- uses the provided keyfile and deployment dir info to invoke the `call_contract` script with the `invalidate-fill` command
- auto-generates meaningful fill IDs for later debugging
-  outputs tx hashes for all sent transactions

**Comment by istankovic** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Development-62442a69c5e6e60010e31cf3/issues/beamer-bridge/beamer/1646?planning-poker) with Zenhub @fredo @GabrielBuragev

**Comment by fredo** [comment]
How do you address the issue with the token? It would be good to have a configurable token. I can imagine one of the following

* add a token option and define by symbol. The script could read the state files and resolve the symbol to the correct address (I already have code for that). 
* Add a separate token.json file to the artifacts dir, which in the future will represent our mintable test token. The script could expect a `token.json` which includes a mapping from chainid to token address. For now we could build a token.json for USDC but in the future when we have the infra it could be our mintable token.

**Comment by istankovic** [comment]
> How do you address the issue which token to use? It would be good to have a configurable token. I can imagine one of the following
> 
>     * add a token option and define by symbol. The script could read the state files and resolve the symbol to the correct address (I already have code for that).
> 
>     * Add a separate token.json file to the artifacts dir, which in the future will represent our mintable test token. The script could expect a `token.json` which includes a mapping from chainid to token address. For now we could build a token.json for USDC but in the future when we have the infra it could be our mintable token.

[...]

I spent quite a while thinking about this. At first, I immediately liked the idea of just reusing the config state files, because... why not? They are already there, contain the necessary info, we can then just reference the token via its symbol etc.

But then after thinking some more about it, I came to dislike the fact that we're now coupling ourselves (or, rather, the challenge test command) to the configuration state and this would necessitate adding yet another option to the already big options set. (In fact I'm not sure how it would work without some convention of naming the configuration state files and then depending on that convention....)

The idea with a separate `token.json` file is somewhat better in that respect, however, I still don't like the fact that we
- have to maintain that file and
- have to add yet another options

IMO, we already have a number of different input and output files so we should really think thrice before adding another kind of files.

So what would be the solution then?

How about we simply provide `--token SYMBOL` option to the `initiate-challenges` command (that's the only one that really needs it, if I'm not mistaken) and then the command figures the token address on its own by querying the chain? Naturally, for each specified chain the `SYMBOL` would be resolved only once and cached for further usage during the runtime of the command. The advantages I see here:
- only one additional option (which we cannot get around anyway if we want this feature)
- no new file types added and no increased maintenance burden

Disadvantages:
- extra RPC calls to resolve `SYMBOL`s to addresses on each chain

I would consider the disadvantage quite acceptable here because we are only doing this very rarely and the time cost is going to be neglibigle if we make sure to use the tokens that were deployed relatively early in the lifetime of the request manager contract.

**Comment by istankovic** [comment]
> > How about we simply provide --token SYMBOL option to the initiate-challenges command (that's the only one that really needs it, if I'm not mistaken) and then the command figures the token address on its own by querying the chain?
> 
> How will the command be able to query the resolution from symbol to address on chain?

By querying the chain, filtering `TokenAdded` events since contract creation until the point the symbol is found.

**Comment by fredo** [comment]
The concept of a symbol does not exist in our contracts. Would you then query the token contract for symbol? If yes, just to be sure that you are aware

Token.symbol() is not part of the ERC20 token standard. So we can not rely that all tokens will implement it. However it is part of the ERC20Detailed interface provided by https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20Detailed

We should be aware that we would bind ourselves to the requirement that tokens implement this function (probably most of them do) and our symbol would have to exactly match what's configured in the token contract.

**Comment by istankovic** [comment]
[...]
> We should be aware that we would bind ourselves to the requirement that tokens implement this function (probably most of them do) and our symbol would have to exactly match what's configured in the token contract.

Definitely, but since here we are talking about our L1 resolution testing commands, I think it's a fair assumption that we use a token that has a symbol. Moreover, if we're to always deploy our own `MintableToken` for just this purpose as we discussed previously, the requirement is trivially satisfied.

**Comment by fredo** [comment]
agree, that's a good point.

**Comment by andfletcher** [comment]
Let make sure that this and the document are aligned . https://docs.google.com/document/d/1_G6u7N7RKvd2Jl2W3TbF1sXkDTqrqjUFZNQ99EvNHZQ/edit
